File: app/src/main/java/com/example/simonastojanovic/memeapp/injection/AppModule.java
package com.example.simonastojanovic.memeapp.injection;

import android.app.Application;
import android.content.Context;

import com.example.simonastojanovic.memeapp.view.MainActivityComponent;

import dagger.Module;
import dagger.Provides;

@Module(subcomponents = {
        MainActivityComponent.class})

public class AppModule {

    @Provides
    Context provideContext(Application application) {
        return application;
    }
}

File: app/src/main/java/com/example/simonastojanovic/memeapp/app/injection/AppModule.kt
package com.example.simonastojanovic.memeapp.app.injection

import android.app.Application
import android.content.Context
import com.example.simonastojanovic.memeapp.app.network.injection.NetworkModule

import dagger.Module
import dagger.Provides

@Module(includes = [NetworkModule::class])
class AppModule {

    @Provides
    internal fun provideContext(application: Application): Context {
        return application
    }
}
